fsharp-backend/src/BackendOnlyStdLib/HttpBytesClient.fs

```diff
@@ -0,0 +1,199 @@
+  handler.ActivityHeadersPropagator <- null
+
+  // Users share the HttpClient, don't let them share cookies!
+  handler.UseCookies <- false
+  handler
+
+let httpClient : HttpClient =
+  let client = new HttpClient(socketHandler, disposeHandler = false)
+  client.Timeout <- System.TimeSpan.FromSeconds 30.0
+  // 100MB seems a reasonable default
+  client.MaxResponseContentBufferSize <- 1024L * 1024L * 100L
+  client
+
+let httpCall
+  (url : string)
+  (queryParams : (string * string list) list)
+  (method : HttpMethod)
+  (reqHeaders : HttpHeaders.T)
+  (reqBody : byte array)
+  : Task<Result<HttpResult, ClientError>> =
+  task {
+    use _ =
+      Telemetry.child
+        "HttpClient.call"
+        [ "request.url", url; "request.method", method ]
+    try
+      let uri = System.Uri(url, System.UriKind.Absolute)
+
+      // currently we only support http(s) requests
+      if uri.Scheme <> "https" && uri.Scheme <> "http" then
+        return Error { url = url; code = 0; error = "Unsupported protocol" }
+      else
+        // Remove the parts of the existing Uri that are duplicated or handled in
+        // other ways
+        let reqUri = System.UriBuilder()
+        reqUri.Scheme <- uri.Scheme
+        reqUri.Host <- uri.Host
+        reqUri.Port <- uri.Port
+        reqUri.Path <- uri.AbsolutePath
+        // Remove the question mark
+        let query =
+          if uri.Query.Length > 0 then String.dropLeft 1 uri.Query else uri.Query
+        reqUri.Query <- HttpQueryEncoding.createQueryString query queryParams
+        use req = new HttpRequestMessage(method, string reqUri)
+        req.Version <- System.Net.HttpVersion.Version30
+        req.Content <- new ByteArrayContent(reqBody)
+
+        // headers - get them before content so we know what to do with content-length
+        let defaultHeaders =
+          Map [ "Accept", "*/*"; "Accept-Encoding", "deflate, gzip, br" ]
+        Map reqHeaders
+        |> Map.mergeFavoringRight defaultHeaders
+        |> Map.iter (fun k v ->
+          // we need this branch, right?
+          if String.equalsCaseInsensitive k "content-type" then
+            try
+              req.Content.Headers.ContentType <-
+                Headers.MediaTypeHeaderValue.Parse(v)
+            with
+            | :? System.FormatException ->
+              Exception.raiseCode "Invalid content-type header"
+          else
+            // Dark headers can only be added once, as they use a Dict. Remove them
+            // so they don't get added twice (eg via Authorization headers above)
+            req.Headers.Remove(k) |> ignore<bool>
+            let added = req.Headers.TryAddWithoutValidation(k, v)
+            // Headers are split between req.Headers and req.Content.Headers so just try both
+            if not added then
+              req.Content.Headers.Remove(k) |> ignore<bool>
+              // CLEANUP: always use lowercase headers
+              req.Content.Headers.Add(k, v))
```

Comment:
Note to self: I ran into an issue here when trying to pass in a request header with an empty key. That's likely why they were previously input as Objs? Though that protection isn't obvious here, and would be good to be clearer.